dpctl/_sycl_event.pyx

```diff
@@ -64,3 +73,132 @@ cdef class SyclEvent:
             SyclEvent cast to a size_t.
         """
         return int(<size_t>self._event_ref)
+
+cdef void _event_capsule_deleter(object o):
+    cdef DPCTLSyclEventRef ERef = NULL
+    if pycapsule.PyCapsule_IsValid(o, "SyclEventRef"):
+        ERef = <DPCTLSyclEventRef> pycapsule.PyCapsule_GetPointer(
+            o, "SyclEventRef"
+        )
+        DPCTLEvent_Delete(ERef)
+
+
+cdef class _SyclEventRaw:
+    """ Python wrapper class for a cl::sycl::event.
+    """
+
+    def __dealloc__(self):
+        DPCTLEvent_Delete(self._event_ref)
+
+
+cdef class SyclEventRaw(_SyclEventRaw):
+    """ Python wrapper class for a cl::sycl::event.
+    """
+
+    @staticmethod
+    cdef void _init_helper(_SyclEventRaw event, DPCTLSyclEventRef ERef):
+        event._event_ref = ERef
+
+    @staticmethod
+    cdef SyclEventRaw _create(DPCTLSyclEventRef eref):
+        cdef _SyclEventRaw ret = _SyclEventRaw.__new__(_SyclEventRaw)
+        SyclEventRaw._init_helper(ret, eref)
+        return SyclEventRaw(ret)
+
+    cdef int _init_event_default(self):
+        self._event_ref = DPCTLEvent_Create()
+        if (self._event_ref is NULL):
+            return -1
+        return 0
+
+    cdef int _init_event_from__SyclEventRaw(self, _SyclEventRaw other):
+        self._event_ref = DPCTLEvent_Copy(other._event_ref)
+        if (self._event_ref is NULL):
+            return -1
+        return 0
+
+    cdef int _init_event_from_SyclEvent(self, SyclEvent event):
+        self._event_ref = DPCTLEvent_Copy(event._event_ref)
+        if (self._event_ref is NULL):
+            return -1
+        return 0
+    
+    cdef int _init_event_from_capsule(self, object cap):
+        cdef DPCTLSyclEventRef ERef = NULL
+        cdef DPCTLSyclEventRef ERef_copy = NULL
+        cdef int ret = 0
+        if pycapsule.PyCapsule_IsValid(cap, "SyclEventRef"):
+            ERef = <DPCTLSyclEventRef> pycapsule.PyCapsule_GetPointer(
+                cap, "SyclEventRef"
+            )
+            if (ERef is NULL):
+                return -2
+            ret = pycapsule.PyCapsule_SetName(cap, "used_SyclEventRef")
+            if (ret):
+                return -2
+            ERef_copy = DPCTLEvent_Copy(ERef)
+            if (ERef_copy is NULL):
+                return -3
+            self._event_ref = ERef_copy
+            return 0
+        else:
+            return -128
+
+    def __cinit__(self,arg=None):
+        cdef int ret = 0
+        if arg == None:
+            ret = self._init_event_default()
+        elif isinstance(arg, _SyclEventRaw):
```

Comment:
We probably want here `type(arg) is _SyclEventRaw`.

Reply:
`isinstance` caters for inheritance while `type` equality checking does not. Don't we need to consider `SyclEventRaw` class when copying?

Reply:
We do not do so for other classes, `SyclDevice`, `SyclQueue`, etc.

Perhaps we need to introduce `.copy()` methods to all of them.